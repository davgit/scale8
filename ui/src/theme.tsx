import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
