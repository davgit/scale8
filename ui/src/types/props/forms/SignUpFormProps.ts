import { FormProps } from '../../../hooks/form/useFormValidation';
import { ApolloError } from '@apollo/client';
import { RefObject } from 'react';
import HCaptcha from '@hcaptcha/react-hcaptcha';

export type SignUpValues = {
    newPassword: string;
    newPasswordConfirm: string;
    domain: string;
    orgName: string;
    fullName: string;
    email: string;
    agree: boolean;
    CAPTCHAToken: string;
    tempAccessCode: string;
};

export type SignUpFormProps = FormProps<SignUpValues> & {
    gqlError?: ApolloError;
    success: boolean;
    email?: string;
    type: string;
    target: string | undefined;
    loading: boolean;
    qsEmail: string | undefined;
    captcha: RefObject<HCaptcha>;
};
