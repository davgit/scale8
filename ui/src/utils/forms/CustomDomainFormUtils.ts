import { ValidateConfiguration } from '../validators/validateFormValues';
import domainValidator from '../validators/domainValidator';
import { FormProps } from '../../hooks/form/useFormValidation';

export type CustomDomainValues = {
    domain: string;
    certificate: string;
    key: string;
};

export const CustomDomainValidators: ValidateConfiguration<CustomDomainValues>[] = [
    {
        field: 'domain',
        validator: domainValidator,
        error: () => 'Invalid domain',
        ignoreEmpty: true,
    },
];

export type CustomDomainFormProps = FormProps<CustomDomainValues> & {
    cname: string;
};
