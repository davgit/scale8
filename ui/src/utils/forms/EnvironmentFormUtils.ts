import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import urlValidator from '../validators/urlValidator';

export type EnvVariable = { key: string; value: string };

export type EnvironmentValues = {
    name: string;
    url: string;
    revisionId: string;
    variables?: EnvVariable[];
    comments: string;
};

const EnvironmentCreateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

const EnvironmentUpdateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

export { EnvironmentCreateValidators, EnvironmentUpdateValidators };
