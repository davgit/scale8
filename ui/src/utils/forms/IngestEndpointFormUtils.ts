import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import { StorageProviderFields, storageProviderValidators } from '../StorageProviderUtils';
import { IngestSchemaWizard } from '../../gql/generated/globalTypes';

export type IngestEndpointValues = StorageProviderFields & {
    name: string;
    wizard: IngestSchemaWizard | '';
    analyticsEnabled: boolean;
};

const IngestEndpointValidators: ValidateConfiguration<IngestEndpointValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Ingest Endpoint name too short',
    },
    ...storageProviderValidators,
];

export { IngestEndpointValidators };
