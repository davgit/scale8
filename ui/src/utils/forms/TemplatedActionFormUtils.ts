import { PlatformDataMapInput } from '../../types/DataMapsTypes';
import { PlatformActionPermissionInput } from '../../types/ActionPermissionsTypes';
import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import descriptionValidator from '../validators/descriptionValidator';
import requiredStringValidator from '../validators/requiredStringValidator';
import parsedCodeValidator from '../validators/parsedCodeValidator';
import { PlatformActionPermissionRequest } from '../../gql/generated/globalTypes';

export type TemplatedActionsValues = {
    name: string;
    description: string;
    icon: string;
    code: string;
    execRaw: boolean;
    platformDataMaps: PlatformDataMapInput[];
    permissionRequests: PlatformActionPermissionInput[];
};

const TemplatedActionValidators: ValidateConfiguration<TemplatedActionsValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Action name too short',
    },
    {
        field: 'description',
        validator: descriptionValidator,
        error: () => 'Description too short',
    },
    {
        field: 'name',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: parsedCodeValidator,
        error: (result) => `Invalid Code${result === 0 || `, Error: ${result}`}`,
    },
    {
        field: 'code',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as string).includes('success()') ? -1 : 0;
        },
        error: () => 'Action code must invoke "success()"',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'description',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'permissionRequests',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as PlatformActionPermissionInput[]).find((_) => {
                if (
                    _.active &&
                    (_.permission === PlatformActionPermissionRequest.READ_PAGE_URL ||
                        _.permission === PlatformActionPermissionRequest.READ_REFERRER_URL)
                ) {
                    return _.urlParts === undefined || _.urlParts.length === 0;
                }
                return false;
            }) === undefined
                ? -1
                : 0;
        },
        error: () => 'Select at least one.',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
];

export { TemplatedActionValidators };
