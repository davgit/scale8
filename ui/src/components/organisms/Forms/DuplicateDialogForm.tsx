import { FC } from 'react';
import { Box, DialogContent, DialogContentText } from '@mui/material';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import FormGqlError from '../../atoms/FormGqlError';
import { DuplicateFormProps } from '../../../utils/forms/DuplicateDialogFormUtils';
import DialogActionsWithCancel from '../../molecules/DialogActionsWithCancel';
import { useTheme } from '@mui/styles';

export type DuplicateDialogProps = DuplicateFormProps & {
    oldName: string;
    handleDialogClose: (checkChanges: boolean) => void;
    description: string;
};

const DuplicateDialogForm: FC<DuplicateDialogProps> = (props: DuplicateDialogProps) => {
    const theme = useTheme();

    return (
        <Box
            component="form"
            sx={{ display: 'flex', flexDirection: 'column' }}
            onSubmit={props.handleSubmit}
        >
            <DialogContent>
                <DialogContentText id="alert-dialog-description">
                    {props.description}
                </DialogContentText>
                <FormGqlError error={props.gqlError} />
                <ControlledTextInput
                    name="name"
                    label="New Name"
                    formProps={props}
                    sx={{
                        width: '100%',
                        margin: theme.spacing(2, 0, 1),
                    }}
                    required
                    autoFocus
                />
            </DialogContent>
            <DialogActionsWithCancel
                disableSubmit={props.isSubmitting}
                handleDialogClose={props.handleDialogClose}
                confirmText={props.submitText}
                ignoreChanges
            />
        </Box>
    );
};

export default DuplicateDialogForm;
