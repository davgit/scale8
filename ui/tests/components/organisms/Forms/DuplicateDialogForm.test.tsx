import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import DuplicateDialogForm from '../../../../src/components/organisms/Forms/DuplicateDialogForm';
import { DuplicateValues } from '../../../../src/utils/forms/DuplicateDialogFormUtils';

const MockForm: FC = () => {
    const formProps = useMockFormProps<DuplicateValues>({
        name: '',
    });

    return (
        <DuplicateDialogForm
            {...formProps}
            oldName=""
            handleDialogClose={() => {
                /*Idle*/
            }}
            description=""
        />
    );
};

describe('DuplicateDialogForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
