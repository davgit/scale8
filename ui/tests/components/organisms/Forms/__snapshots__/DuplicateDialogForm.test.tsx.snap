// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DuplicateDialogForm form render unchanged 1`] = `
<DocumentFragment>
  <form
    class="MuiBox-root css-j7qwjs"
  >
    <div
      class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
    >
      <p
        class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-s2fmnb-MuiTypography-root-MuiDialogContentText-root"
        id="alert-dialog-description"
      />
      <div
        class="MuiFormControl-root MuiTextField-root css-1rrfi79-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
        >
          New Name
          <span
            aria-hidden="true"
            class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
          >
            â€‰*
          </span>
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiDialogActions-root MuiDialogActions-spacing css-snbeu5-MuiDialogActions-root"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-7kx4b6-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Cancel
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</DocumentFragment>
`;
